zap/src/main/java/org/zaproxy/zap/extension/history/SiteNotesAddDialog.java

```diff
@@ -244,7 +246,7 @@ private JButton getBtnOk() {
             btnOk.addActionListener(
                     e -> {
                         HistoryReference historyRef = childSiteNodes.getSelectedValue();
-                        historyRef.setNote(getTxtDisplay().getText());
+                        if (historyRef != null) historyRef.setNote(getTxtDisplay().getText());
```

Comment:
You may have seen this elsewhere in older bits of ZAP code but please do use braces even for single statements, it makes things more clear.

Reply:
Up at 218 as well.